src/main/java/com/avispl/symphony/dal/infrastructure/management/tripleplay/TriplePlayAggregatorCommunicator.java

```diff
@@ -0,0 +1,714 @@
+
+	/**
+	 * cachedClients store all client to update information of clients
+	 */
+	private TreeMap<String, Client> cachedClients = new TreeMap<>();
+
+	/**
+	 * cachedAggregatedDevices store all AggregatedDevice to map and show information of clients
+	 */
+	private Map<String, AggregatedDevice> cachedAggregatedDevices = new ConcurrentHashMap<>();
+
+	/**
+	 * cachedServices store all service
+	 */
+	private HashMap<String, Service> cachedServices = new HashMap<>();
+
+	private boolean isEmergencyDelivery = false;
+	private ObjectMapper objectMapper = new ObjectMapper();
+
+	/**
+	 * Retrieves {@link #pollingInterval}
+	 *
+	 * @return value of {@link #pollingInterval}
+	 */
+	public String getPollingInterval() {
+		return pollingInterval;
+	}
+
+	/**
+	 * Sets {@link #pollingInterval} value
+	 *
+	 * @param pollingInterval new value of {@link #pollingInterval}
+	 */
+	public void setPollingInterval(String pollingInterval) {
+		this.pollingInterval = pollingInterval;
+	}
+
+	@Override
+	public List<Statistics> getMultipleStatistics() throws Exception {
+		//Because there are some threads that keep running when the next getMultiple is called,
+		// so we have to stop all those threads just before the next getMultiple runs
+		if (executorService != null) {
+			for (Future future : clientExecutionPool) {
+				future.cancel(true);
+			}
+			clientExecutionPool.clear();
+		}
+		reentrantLock.lock();
+		try {
+			if (!isEmergencyDelivery) {
+				int currentSizeCacheClients = cachedClients.size();
+				retrieveClients();
+				retrieveServices();
+				localPollingInterval = calculatingLocalPollingInterval();
+				deviceStatisticsCollectionThreads = calculatingThreadQuantity();
+				//Multi thread for get information of client
+				retrieveInformationOfAllClients(currentSizeCacheClients);
+			}
+			isEmergencyDelivery = false;
+		} finally {
+			reentrantLock.unlock();
+		}
+		return null;
+	}
+
+	@Override
+	public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+		String property = controllableProperty.getProperty();
+		String value = String.valueOf(controllableProperty.getValue());
+		String deviceId = controllableProperty.getDeviceId();
+		reentrantLock.lock();
+		try {
+			if (StringUtils.isNotNullOrEmpty(deviceId) && cachedAggregatedDevices.containsKey(deviceId)) {
+				aggregatedDeviceControl(property, value, cachedAggregatedDevices.get(deviceId));
+			}
+
+		} finally {
+			reentrantLock.unlock();
+		}
+	}
+
+	@Override
+	public void controlProperties(List<ControllableProperty> list) throws Exception {
+		for (ControllableProperty controllableProperty : list) {
+			controlProperty(controllableProperty);
+		}
+	}
+
+	@Override
+	protected void authenticate() throws Exception {
+
+	}
+
+	@Override
+	public List<AggregatedDevice> retrieveMultipleStatistics() throws Exception {
+		if (logger.isWarnEnabled()) {
+			logger.warn("Start call retrieveMultipleStatistic");
```

Comment:
Is warning level necessary here? This is what would happen normally, so maybe warning is a bit too much :)

Reply:
I fixed it